### PR TITLE
Release notes in documentation

### PR DESCRIPTION
**Pull request by agriffard**
Related to #5841

### CODE REVIEW

**Comment by agriffard** [comment]
https://docs.orchardcore.net/en/ag-releasenotes/docs/releases/1.0.0-rc1/

**Comment by agriffard** [comment]
image

**Comment by sebastienros** [comment]
Add the date of the release

**Comment by agriffard** [comment]
image

File: src/docs/releases/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Releases
```
Comment (Piedone) [inline src/docs/releases/README.md:1]: Will you add links here, or is it auto-generated somehow?

File: src/docs/releases/1.0.0-rc1.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
+# Orchard Core 1.0.0-rc1
```
Comment (Piedone) [inline src/docs/releases/1.0.0-rc1.md:1]: As I've suggested under https://github.com/OrchardCMS/OrchardCore/issues/5841 the O1 release notes format I think is pretty good. While it's not entirely applicable a pretty much mechanical release notes like this is not sufficient I think: The long list of enhancements might be a good reference but if we want people to get a general understanding of what's new it won't work.

While I don't see a point on going back to betas and doing this but eventually we need to start and probably do that with RC1? The structure I can imagine working is what I've added here: https://github.com/OrchardCMS/OrchardCore/blob/62c218ae7ed7d3354db66e588e5ee2652c9ae2be/src/docs/topics/publishing-releases/README.md "Create release notes..."

File: src/docs/releases/README.md (hunk 1 above)
Comment (agriffard) [reply src/docs/releases/README.md:1]: I guess I can remove it if we don't use it. 
Not auto-generated.

File: src/docs/releases/1.0.0-rc1.md (hunk 2 above)
Comment (agriffard) [reply src/docs/releases/1.0.0-rc1.md:1]: Yes, we can add more details to the release notes than just the generated PRs links:
- Breaking Changes / Migration
- Enhancements / New features
- Pre-requesites (Framework version)

Comment (Piedone) [reply src/docs/releases/1.0.0-rc1.md:1]: So like this?

>   - Overview of the release's highlights and goals. What do you want people to remember this release for?
>  - Prerequisites. What framework version do you need, anything else to work with Orchard?
>  - Upgrade steps, any migration necessary from previous versions.
>  - Full changelog. You can generate this with [github-changelog](https://github.com/cfpb/github-changelog) with the `changelog OrchardCMS OrchardCore <previous version> <current version>` command, e.g. `changelog OrchardCMS OrchardCore 1.0.0-rc1 1.0.0-rc2`.

Comment (agriffard) [reply src/docs/releases/1.0.0-rc1.md:1]: Exactly.

**Comment by agriffard** [comment]
Here is an app I made in order to generate the PRs links (because the `changelog` only return #1234 without a link):
https://github.com/agriffard/Changelog.OrchardCore

File: mkdocs.yml [hunk 3]
```diff
@@ -164,4 +164,10 @@ nav:
         - Learning: docs/resources/README.md
         - Workshops: docs/resources/workshops/README.md
         - Branding: docs/resources/branding/README.md
+    - Releases: 
+        - 1.0.0-rc2: docs/releases/1.0.0-rc2.md
```
Comment (hishamco) [inline mkdocs.yml:168]: I think the release should ordered by version ascending

**Pull request closed without merging**